pkg/operator/ceph/config/config.go

```diff
@@ -138,6 +138,21 @@ func SetOrRemoveDefaultConfigs(
 		}
 	}
 
+	// remove `rbd_default_features` since it is no longer used in the
+	// upgraded clusters. This was set to `3` previously from Rook.
+	rbdDefaultFeatures := "rbd_default_features"
+	val, err := monStore.Get("mon", rbdDefaultFeatures)
```

Comment:
```suggestion
	val, err := monStore.Get("global", rbdDefaultFeatures)
```
as Rook's override was global (and you are passing "global" to `Delete` below)

Reply:
>sh-4.4$ ceph config get global rbd_default_features
Error EINVAL: unrecognized entity 'global'


@idryomov get for `global` failed for me. that why I went for `mon`

>hypothetically, a user could have manually set rbd_default_features to 3 in Ceph (i.e. behind Rook's back) and you would delete their setting here

Yes, I agree on this one.

>given that pretty much all Ceph CSI deployments define an explicit list of image features that is passed to rbd_create API in the storage class, thus effectively overriding Rook's setting, who would benefit from this change?

This is to make sure upgraded clusters work fine without any image features set in the storageclass. As Rook might have added it global setting in the previous version and we don't have any automation to remove it ( at least thinking from downstream point of view).  As we are planning to make imageFeatues optional in storageclass without this one upgrade clusters still defaulted to `3`.

Reply:
@idryomov Without this change, we cannot use https://github.com/red-hat-storage/ocs-operator/pull/1512#issuecomment-1042636105? if we do so we might not get the expected behavior.

Reply:
> @idryomov get for `global` failed for me. that why I went for `mon`

This is wrong because `mon` can have its own, non-global, override.  You need to use `ceph config dump --format=json` and search for a respective `"section": "global"` entry in the resulting array.  (Previously I said that this was an omission in Ceph, but, after looking at the code, `ceph config get` works as intended.  The fact that it isn't symmetric to `ceph config set` is unfortunate but unlikely to be fixed.)

Reply:
Yes, that's correct @idryomov. i will make the requested suggestions will keep doc only PR for upstream and will have downstream only patch for downstream

Reply:
> > @idryomov get for `global` failed for me. that why I went for `mon`
> 
> This is wrong because `mon` can have its own, non-global, override. You need to use `ceph config dump --format=json` and search for a respective `"section": "global"` entry in the resulting array. (Previously I said that this was an omission in Ceph, but, after looking at the code, `ceph config get` works as intended. The fact that it isn't symmetric to `ceph config set` is unfortunate but unlikely to be fixed.)

I have attempted to clarify this in https://github.com/ceph/ceph/pull/45219.  Hopefully it makes things clearer!